lib/facility.rb

```diff
@@ -1,14 +1,65 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil if !@services.include?("Vehicle Registration")
+    if @services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Date.today
+      if vehicle.plate_type == :antique
+        @collected_fees += 25
+        elsif vehicle.plate_type == :ev
+        @collected_fees += 200
+        elsif vehicle.plate_type == :regular
+        @collected_fees += 100
+      end
```

Comment:
1. Keep an eye on your indentation
2. Conventionally, every `if` block with an `elsif` should also have an `else` as a catch-all. In this case, I'd suggest
```
if vehicle.plate_type == :antique
  @collected_fees += 25
elsif vehicle.plate_type == :ev
  @collected_fees += 200
else # all other vehicles are regular. Else doesn't take a condition.
  @collected_fees += 100
end
```